### PR TITLE
Updating Functions ARM APIs

### PR DESCRIPTION
**Pull request by mathewc**
Merging the changes made in https://github.com/Azure/azure-rest-api-specs/pull/7174 to the latest spec version. No service changes have been made - just pulling these previously reviewed specs forward. No backend RP changes were made - this just updates the spec to be inline with what is already there.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/WebApps.json [hunk 1]
```diff
@@ -3323,14 +3323,14 @@
...
-      "post": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/functions/{functionName}/keys/{keyName}": {
+      "put": {
         "tags": [
           "WebApps"
         ],
-        "summary": "Get function secrets for a function in a web site, or a deployment slot.",
-        "description": "Description for Get function secrets for a function in a web site, or a deployment slot.",
-        "operationId": "WebApps_ListFunctionSecrets",
+        "summary": "Add or update a function secret.",
```
Comment (yungezz) [inline specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/WebApps.json]: if path name changed `secret` to `key`, description,. operationId should be changed also for consistent.

File: specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/WebApps.json [hunk 2]
```diff
@@ -3323,14 +3323,14 @@
         }
       }
     },
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/functions/{functionName}/listsecrets": {
-      "post": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/functions/{functionName}/keys/{keyName}": {
```
[the remark this answers is not in the document]
Comment (mathewc) [reply specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/WebApps.json]: This is just a diff/whitespace issue. If you look at this in windiff for example you'll see no existing routes were changed.

File: specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/WebApps.json (hunk 1 above)
Comment (mathewc) [reply specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/WebApps.json]: No paths or routes were changed in this PR. Please see above.

**Comment by yungezz** [comment]
communicated with @mathewc , this is a swagger bug fixing to align with existing service behavior, the change is reviewed in another PR  #7174  which is for another version. so no user breaking, so approve this.

**Review by yungezz — APPROVED** [review]

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
looking at LintDiff error with @mathewc

**Comment by mathewc** [comment]
Catherine, I don’t understand these failures. The spec changes I’m making here were already accepted and checked in previously. The Lint error in question appears to be from the definition of the KeyInfo type and the fact that it isn’t a ProxyOnlyResource. As you can see from the previous stable version [here](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/WebApps.json), the definition is the same. These APIs/definitions were already merged previously by you in this PR.

Now I could just go in and make this spec definition inherit ProxyOnlyResource properties, but that would be incorrect – those properties are never accepted/returned from the APIs using this type. I think we need to get this in as is – please tell me how to proceed.

"KeyInfo": {
  "description": "Function key info.",
  "type": "object",
  "properties": {
              "name": {
                "description": "Key name",
                "type": "string"
              },
              "value": {
                "description": "Key value",
                "type": "string"
              }
  }
}

**Comment by mathewc** [comment]
Lint errors fixed @yungezz

**Review by yungezz — APPROVED** [review]

**Pull request merged**